internal/controller/components/kserve/kserve_controller_actions.go

```diff
@@ -176,3 +180,58 @@ func versionedWellKnownLLMInferenceServiceConfigs(_ context.Context, version str
 	}
 	return nil
 }
+
+// checkPreConditions checks if there are optional operators that KServe could use.
+func checkPreConditions(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	k, ok := rr.Instance.(*componentApi.Kserve)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentApi.Kserve)", rr.Instance)
+	}
+	dsc, err := cluster.GetDSC(ctx, rr.Client)
+	if err != nil {
+		return fmt.Errorf("failed to get DSC: %w", err)
+	}
+	if dsc.Spec.Components.Kserve.ManagementState != operatorv1.Managed {
+		return nil
+	}
```

Comment:
nip: this is probably not required as the `Kserve` CR would only exists if `ManagmentState == Managed`